File: app/src/main/java/com/wolcano/musicplayer/music/content/Binder.java
package com.wolcano.musicplayer.music.content;

import android.app.Activity;
import android.view.View;

import com.wolcano.musicplayer.music.mvp.listener.Bind;

import java.lang.reflect.Field;

public class Binder {

    public static void bindIt(Activity activity) {
        bindIt(activity, activity.getWindow().getDecorView());
    }
    public static void bindIt(Object target, View source) {
        Field[] fieldArr = target.getClass().getDeclaredFields();
        if (fieldArr.length > 0 && fieldArr != null) {
            for (Field f : fieldArr) {
                try {
                    f.setAccessible(true);
                    if (f.get(target) != null) {
                        continue;
                    }
                    Bind bind = f.getAnnotation(Bind.class);
                    if (bind != null) {
                        int id = bind.value();
                        f.set(target, source.findViewById(id));
                    }
                } catch (Exception e) {
                    e.printStackTrace();
                }
            }
        }
    }
}

File: app/src/main/java/com/wolcano/musicplayer/music/content/Binder.kt
package com.wolcano.musicplayer.music.content

import android.app.Activity
import android.view.View
import com.wolcano.musicplayer.music.mvp.listener.Bind

object Binder {

    fun bindIt(activity: Activity) {
        bindIt(activity, activity.window.decorView)
    }

    fun bindIt(target: Any, source: View?) {
        val fieldArr = target.javaClass.declaredFields
        if (fieldArr.isNotEmpty()) {
            for (f in fieldArr) {
                try {
                    f.isAccessible = true
                    if (f[target] != null) {
                        continue
                    }
                    val bind = f.getAnnotation(Bind::class.java)
                    if (bind != null) {
                        val id: Int = bind.value
                        f[target] = source?.findViewById(id)
                    }
                } catch (e: Exception) {
                    e.printStackTrace()
                }
            }
        }
    }

}